torch/nn/parameter.pyi

```diff
@@ -1,29 +1,47 @@
-# mypy: allow-untyped-defs
 from typing_extensions import TypeGuard
 
-from torch import device, dtype, Tensor
+from torch import dtype, Tensor
+from torch._C import _TensorMeta
+from torch.types import Device
 
-class Parameter(Tensor):
-    def __init__(self, data: Tensor = ..., requires_grad: bool = ...) -> None: ...
+class _ParameterMeta(_TensorMeta):
+    def __instancecheck__(self, instance: object) -> bool: ...
+
+class Parameter(Tensor, metaclass=_ParameterMeta):
+    def __init__(
+        self,
+        data: Tensor | None = None,
+        requires_grad: bool = True,
+    ) -> None: ...
 
 def is_lazy(
     param: Tensor,
 ) -> TypeGuard[UninitializedParameter | UninitializedBuffer]: ...
 
-class UninitializedParameter(Tensor):
-    def __init__(self, data: Tensor = ..., requires_grad: bool = ...) -> None: ...
+class UninitializedParameter(Parameter):
+    def __init__(
+        self,
+        requires_grad: bool = True,
+        device: Device = None,
+        dtype: dtype | None = None,
+    ) -> None: ...
     def materialize(
         self,
         shape: tuple[int, ...],
-        device: device | None = None,
+        device: Device = None,
         dtype: dtype | None = None,
     ) -> None: ...
 
 class UninitializedBuffer(Tensor):
-    def __init__(self, data: Tensor = ..., requires_grad: bool = ...) -> None: ...
+    def __init__(
+        self,
+        requires_grad: bool = False,
+        device: Device = None,
+        dtype: dtype | None = None,
+    ) -> None: ...
     def materialize(
         self,
         shape: tuple[int, ...],
-        device: device | None = None,
+        device: Device = None,
```

Comment:
This is actually optional!

Reply:
`Device` already includes `None`.

```python
type Device = device | str | int | None
```

https://github.com/pytorch/pytorch/blob/10c7f037fe3271cb3865816c216007ba403f5347/torch/types.py#L50